### PR TITLE
Removed _dot and norm2, as they are not (or versions of them) in Base

### PR DESCRIPTION
**Pull request by pkofod**
As in commit message, I've removed _dot and norm2, as they are now in Base (as vecdot and vecnorm), and norm2 is not even used .

### CODE REVIEW

**Comment by mlubin** [comment]
Fails on 0.3. You can bump REQUIRE and update the travis script to run on 0.4 and nightly only.

**Comment by pkofod** [comment]
I bumped to v0.4, but I might as well remove the unnecessary @compat macros in this PR as well then. I'll do it later.

**Comment by tkelman** [comment]
If this gets tagged, it should be as 0.5.0.

**Comment by pkofod** [comment]
I forgot to include a file in the last commit, and I also fixed a typo in the commit message. @tkelman ok.

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-julia 0.3
+julia 0.4
 Calculus
 DualNumbers
 Compat 0.7.1
```
Comment (mlubin) [inline REQUIRE]: You could remove the dependency on ``Compat`` while you're at it

**Comment by johnmyleswhite** [comment]
I'm good with this. If we break things for 0.4, it would be good to make some of the other breaking changes at the same time.

**Comment by johnmyleswhite** [comment]
I tagged a release that still works on 0.3, so let's get to breaking stuff

**Comment by johnmyleswhite** [comment]
Sigh. This needs a rebase. Sorry for that.

**Comment by pkofod** [comment]
No problem, I'll fix it asap. Makes sense to tag a "last" v0.3 compatible version before merging this.

**Comment by johnmyleswhite** [comment]
Yeah, that's what I did.

**Comment by pkofod** [comment]
Yes I know, my English was probably a bit unclear. Maybe I should have written:
"Tagging a "last" v0.3 compatible version before merging this was smart."

:)

**Comment by johnmyleswhite** [comment]
Sorry for the confusion. I should have thought more carefully about what you said.

**Pull request merged**